src/realm/cluster.cpp

```diff
@@ -2174,63 +2212,99 @@ ObjKey ClusterTree::ConstIterator::load_leaf(ObjKey key) const
     }
 }
 
-ClusterTree::ConstIterator::pointer Table::ConstIterator::operator->() const
+auto ClusterTree::ConstIterator::operator[](size_t n) -> reference
 {
     if (m_storage_version != m_tree.get_storage_version(m_instance_version)) {
-        ObjKey k = load_leaf(m_key);
-        if (k != m_key)
-            throw std::out_of_range("Object was deleted");
+        // reload
```

Comment:
Would it make sense to implement this in terms of "operator+=". Clone, op+=, then dereference ?

Reply:
I don't understand this proposal. The object pointed to should not be changed.

Reply:
I'm referring to implementing operator[] by
it::op[idx]:
 new_it = it.clone()
 new_it += idx
 return *new_it